### PR TITLE
Multi vendor support

### PR DESCRIPTION
**Pull request by ghost**
This PR adds Intel,Dell,Lenova,Marvel,Nivida,AMD and Mellanox to the accepted vendors in the BIOS vendor field check. 

It also resolves a bug where 'Not Provided' is returned as a value from the dmi table.

### CODE REVIEW

**Comment by ghost** [comment]
Hi @glimchb, I was advised your the right person to speak to about this PR please let me know if its suitable or if there are any further changes to be checked. I tried to run the tests but it looks like they require Google specific pip dependency's.  I did test this on Intel hardware.

**Comment by glimchb** [comment]
> Hi @glimchb, I was advised your the right person to speak to about this PR please let me know if its suitable or if there are any further changes to be checked. I tried to run the tests but it looks like they require Google specific pip dependency's. I did test this on Intel hardware.

thank you @AdaRogerson  !

please sign-off both of your commits, so it can pass "DCO" checker and we can merge it
`git commit -s`

File: smbios_validation_tool/constants.py [hunk 1]
```diff
@@ -72,3 +72,4 @@ class FieldValueRegexps(enum.Enum):
   SOCKET_DESIGNATION_REGEXP = r'(CPU|P)\d+'
   DEVICE_LOCATOR_REGEXP = r'(?:[^\d]*)\d+'
   BANK_LOCATOR_REGEXP = r'.*(Node|Channel).*'
+  VENDORS_REGEXP = r'(?i).*(Intel|Google|Dell|Lenova|Marvel|Nivida|AMD|Mellanox).*'
```
Comment (seroyer) [inline smbios_validation_tool/constants.py]: I assume that's supposed to be "Lenovo", "Marvell", and "Nvidia"?  What should the value be for HPE?  Should we limit this field at all?  Or should we just allow "*"?

**Review by seroyer — CHANGES_REQUESTED** [review]

File: smbios_validation_tool/constants.py (hunk 1 above)
Comment (glimchb) [reply smbios_validation_tool/constants.py]: I'm ok with `*`

Comment (ghost) [reply smbios_validation_tool/constants.py]: @seroyer thanks for the heads up on the typos. 
I am not sure what todo about HPE, there is some onus on the vendors to add the values for there equipment but i am unsure what they label there own stuff as.

**Review by glimchb — APPROVED** [review]
I'm ok with either this implementation or using `*` instead of explicit list of vendors
approved

File: smbios_validation_tool/constants.py (hunk 1 above)
Comment (seroyer) [reply smbios_validation_tool/constants.py]: I'm ok with this, thanks.  I think we'll want to decide down the road if we want this to be exclusive to OPI qualified partners (not necessarily OPI members, those that have integration/release tests pass), or if we should just open it up to anything.

**Review by seroyer — APPROVED** [review]

**Pull request merged**